contracts/internal/EverlongPositions.sol

```diff
@@ -0,0 +1,296 @@
+                _position.bondAmount,
+                _minCloseLongOutput(
+                    _position.maturityTime,
+                    _position.bondAmount
+                ),
+                IHyperdrive.Options(address(this), _asBase, "")
+            );
+
+            // Update positions to reflect the newly closed long.
+            _handleCloseLong(uint128(_position.bondAmount));
+        }
+    }
+
+    // PERF: Popping then pushing the same position is inefficient.
+    /// @dev Account for closed bonds at the oldest `maturityTime`
+    ///      within the `PositionManager`.
+    /// @param _bondAmountClosed Amount of bonds closed.
+    function _handleCloseLong(uint128 _bondAmountClosed) internal {
+        // Remove the oldest position from the front queue.
+        Position memory _position = _decodePosition(_positions.popFront());
+
+        // Compare the input bond amount to the most mature position's
+        // `bondAmount`.
+        if (_bondAmountClosed > _position.bondAmount) {
+            revert InconsistentPositionBondAmount();
+        }
+        // The amount to close equals the position size.
+        // Nothing further needs to be done.
+        else if (_bondAmountClosed == _position.bondAmount) {
+            emit PositionClosed(_position.maturityTime);
+        } else {
+            // The amount to close is not equal to the position size.
+            // Push the position less the amount of longs closed to the front.
+            _positions.pushFront(
+                _encodePosition(
+                    _position.maturityTime,
+                    _position.bondAmount - _bondAmountClosed
+                )
+            );
+            emit PositionUpdated(
+                _position.maturityTime,
+                _position.bondAmount - _bondAmountClosed,
+                0
+            );
+        }
+    }
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Position Encoding/Decoding                              │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @dev Encodes the `_maturityTime` and `_bondAmount` into bytes32
+    ///      to store in the queue.
+    /// @param _maturityTime Timestamp the position matures.
+    /// @param _bondAmount Amount of bonds in the position.
+    /// @return The bytes32 encoded `Position`.
+    function _encodePosition(
+        uint128 _maturityTime,
+        uint128 _bondAmount
+    ) public pure returns (bytes32) {
+        return bytes32((uint256(_maturityTime) << 128) | uint256(_bondAmount));
+    }
+
+    /// @dev Decodes the bytes32 data into a `Position` struct.
+    /// @param _position The bytes32 encoded data.
+    /// @return The decoded `Position`.
+    function _decodePosition(
+        bytes32 _position
+    ) public pure returns (Position memory) {
+        uint128 _maturityTime = uint128(bytes16(_position << 128));
+        uint128 _bondAmount = uint128(bytes16(_position));
```

Comment:
Hmm, looks like we updated the other one, but not this one. I still think this is going to be more expensive than something like this:

```suggestion
        uint128 _maturityTime = uint128(uint256(_position) >> 128);
        uint128 _bondAmount = uint128(uint256(_position));
```